src/encode.cpp

```diff
@@ -381,6 +381,70 @@ optional<string> encodeOp(State &st, mlir::tensor::FromElementsOp op) {
   return {};
 }
 
+template<>
+optional<string> encodeOp(State &st, mlir::tensor::ExtractSliceOp op) {
+  vector<Expr> offsets, sizes, strides;
+  auto src = st.regs.get<Tensor>(op.getOperand(0));
+  auto srcType = op.getOperand(0).getType().dyn_cast<mlir::ShapedType>();
+  auto res = op.getResult();
+  auto resType = res.getType().dyn_cast<mlir::ShapedType>();
+
+  auto varIdx = 0;
+  for (auto s: extractFromI64ArrayAttr(op.static_offsets())){
+    if(s ==-0x8000000000000000)
+      offsets.push_back(st.regs.get<Index>(op.offsets()[varIdx++]));
+    else
+      offsets.push_back(Index(s));
+  }
+  varIdx = 0;
+  for (auto s: extractFromI64ArrayAttr(op.static_sizes())) {
+    if(s ==-0x8000000000000000)
+      sizes.push_back(st.regs.get<Index>(op.sizes()[varIdx++]));
+    else
+      sizes.push_back(Index(s));
+  }
+  varIdx = 0;
+  for(auto s: extractFromI64ArrayAttr(op.static_strides())) {
+    if(s ==-0x8000000000000000)
+      strides.push_back(st.regs.get<Index>(op.strides()[varIdx++]));
+    else
+      strides.push_back(Index(s));
+  }
+
+  vector<smt::Expr> dims; 
+
+  unsigned j=0;
+  for (unsigned i = 0; i < resType.getRank(); i++) {
+    while((sizes[j] == Index(1)).isTrue()) {
+      j++;
+    }
+
+    // check if output tensor matches size
+    assert((sizes[j] == Index(resType.getDimSize(i))).isTrue());
```

Comment:
Is this still true if the output dimension is not constant?

Reply:
added a case when dimension size is unknown